packages/react/src/components/sideNavigation/SideNavigation.tsx

```diff
@@ -80,9 +81,11 @@ export const SideNavigation = ({
   const isMobile = useMobile();
   const shouldRenderMenu = !(isMobile && !mobileMenuOpen);
 
-  const mainLevels = React.Children.map(children, (child, index) => {
+  const childElements = getChildrenAsArray(children);
+
+  const mainLevels = childElements.map((child, index) => {
     if (isValidElement(child) && (child.type as FCWithName).componentName === 'MainLevel') {
-      return cloneElement(child, { index });
+      return cloneElement(child, { key: index, index });
```

Comment:
The index key value pair here gives an TS error on my computer, does it display to you?
"No overload matches this call.
  The last overload gave the following error.
    Argument of type '{ key: number; index: number; }' is not assignable to parameter of type 'Partial<unknown> & Attributes'.
      Object literal may only specify known properties, and 'index' does not exist in type 'Partial<unknown> & Attributes'.ts(2769)"

Reply:
Interestingly it does not do that for me 🤔